modules/xfeatures2d/include/opencv2/xfeatures2d.hpp

```diff
@@ -945,6 +945,40 @@ class CV_EXPORTS AffineFeature2D : public Feature2D
         bool useProvidedKeypoints=false ) = 0;
 };
 
+/**
+@brief Class implementing the Tree Based Morse Regions (TBMR) as described in
+@cite Najman2014 extended with scaled extraction ability.
+
+@param min_area prune areas smaller than minArea
+@param max_area_relative prune areas bigger than maxArea = max_area_relative *
+input_image_size
+@param scale_factor scale factor for scaled extraction.
+@param n_scales number of applications of the scale factor (octaves).
+
+@note This algorithm is based on Component Tree (Min/Max) as well as MSER but
+uses a Morse-theory approach to extract features.
+
+Features are ellipses (similar to MSER, however a MSER feature can never be a
+TBMR feature and vice versa).
+
+*/
+class CV_EXPORTS TBMR : public AffineFeature2D
+{
+public:
+    CV_WRAP static Ptr<TBMR> create(int min_area = 60,
```

Comment:
1. Class itself should be wrapped: `class CV_EXPORTS` => `class CV_EXPORTS_W`
2. Base class should be wrapped too: `class CV_EXPORTS_W AffineFeature2D`

Reply:
ok, done